osbs/build/plugins_configuration.py

```diff
@@ -0,0 +1,583 @@
+        }
+
+        # Remove empty values, and always convert to string for better interaction
+        # with Configuration class and JSON encoding
+        config_kwargs = dict((k, str(v)) for k, v in config_kwargs.items() if v is not None)
+
+        if not self.user_params.build_imagestream.value:
+            config_kwargs['build_image'] = self.user_params.build_image.value
+
+        self.template_set_arg(phase, plugin, 'config_kwargs', config_kwargs)
+
+    def render_resolve_composes(self):
+        phase = 'prebuild_plugins'
+        plugin = 'resolve_composes'
+
+        if not self.template_has_plugin_conf(phase, plugin):
+            return
+
+        if self.user_params.yum_repourls.value:
+            logger.info('removing %s from request as yum_repourls is user_paramsified', plugin)
+            self.remove_plugin(phase, plugin)
+            return
+
+        self.set_plugin_arg_valid(phase, plugin, 'compose_ids',
+                                  self.user_params.compose_ids.value)
+
+        self.set_plugin_arg_valid(phase, plugin, 'signing_intent',
+                                  self.user_params.signing_intent.value)
+
+    def render_resolve_module_compose(self):
+        phase = 'prebuild_plugins'
+        plugin = 'resolve_module_compose'
+
+        if self.template_has_plugin_conf(phase, plugin):
+            if not self.user_params.flatpak.value:
+                self.remove_plugin(phase, plugin)
+                return
+
+            self.set_plugin_arg_valid(phase, plugin, 'compose_ids',
+                                      self.user_params.compose_ids.value)
+
+    def render_squash(self):
+        phase = 'prepublish_plugins'
+        plugin = 'squash'
+
+        if self.user_params.flatpak.value:
+            # We'll extract the filesystem anyways for a Flatpak instead of exporting
+            # the docker image directly, so squash just slows things down.
+            self.remove_plugin(phase, plugin)
+            return
+
+    def render_tag_from_config(self):
+        """Configure tag_from_config plugin"""
+        phase = 'postbuild_plugins'
+        plugin = 'tag_from_config'
+        if not self.has_tag_suffixes_placeholder():
+            return
+
+        unique_tag = self.user_params.image_tag.value.split(':')[-1]
+        tag_suffixes = {'unique': [unique_tag], 'primary': []}
+
+        if self.user_params.build_type.value == BUILD_TYPE_ORCHESTRATOR:
+            if self.user_params.scratch.value:
+                pass
+            elif self.user_params.isolated.value:
+                tag_suffixes['primary'].extend(['{version}-{release}'])
+            else:
+                tag_suffixes['primary'].extend(['latest', '{version}', '{version}-{release}'])
+                if self.user_params.additional_tags:
+                    tag_suffixes['primary'].extend(self.user_params.additional_tags.tags)
+
+        self.template_set_arg(phase, plugin, 'tag_suffixes', tag_suffixes)
+
+    def render(self):
```

Comment:
Parent's version calls `self.spec.validate()`, do the same here for `user_params` ?

Reply:
Fixed.